improver/blending/calculate_weights_and_blend.py

```diff
@@ -179,30 +178,37 @@ def _update_spatial_weights(self, cube, weights, fuzzy_length):
         weights = plugin(cube, weights)
         return weights
 
-    def _update_metadata_only(self, cube, attributes_dict, cycletime):
-        """
-        If blend_coord has only one value (for example cycle blending with
-        only one cycle available), or is not present (case where only
-        one model has been provided for a model blend), update attributes
-        and time coordinates and return.
+    def _remove_zero_weighted_slices(self, cube, weights):
+        """Removes any cube and weights slices where the 1D weighting factor
+        is zero
+
+        Args:
+            cube (iris.cube.Cube):
+                The data cube to be blended
+            weights (iris.cube.Cube):
+                1D cube of weights varying along self.blend_coord
+
+        Returns:
+            (tuple): tuple containing:
+                **cube** (iris.cube.Cube):
+                    Data cube without zero-weighted slices
+                **weights** (iris.cube.Cube):
+                    Weights without zeroes
         """
-        result = cube.copy()
-        if attributes_dict is not None:
-            amend_attributes(result, attributes_dict)
-
-        (result,) = rebadge_forecasts_as_latest_cycle([result], cycletime)
-        if self.blend_coord in ["forecast_reference_time", "model_id"]:
-            for coord in ["forecast_period", "forecast_reference_time"]:
-                msg = f"{coord} will be removed in future and should not be used"
-                result.coord(coord).attributes.update({"deprecation_message": msg})
-
-            if cycletime is not None:
-                add_blend_time(result, cycletime)
-            else:
-                msg = "Current cycle time is required for cycle and model blending"
-                raise ValueError(msg)
+        slice_out_vals = []
+        for wslice in weights.slices_over(self.blend_coord):
+            if np.sum(wslice.data) == 0:
+                slice_out_vals.append(wslice.coord(self.blend_coord).points[0])
 
-        return result
+        if not slice_out_vals:
+            return cube, weights
+
+        constraint = iris.Constraint(
+            coord_values={self.blend_coord: lambda x: x not in slice_out_vals}
+        )
+        cube = cube.extract(constraint)
```

Comment:
Does this create a copy or edit in place? The latter would mean you could avoid returning the cubes

Reply:
I don't know offhand.  The iris documentation says it returns a subset of the cube - so I guess it allocates a new cube rather than modifying?  But I wouldn't know how to figure that out from the documentation / code.